### PR TITLE
Replace TravisCI with GitHub Actions

### PR DESCRIPTION
**Pull request by Turbo87**
Travis CI has gotten pretty slow for open-source projects lately and with things like https://twitter.com/peter_szilagyi/status/1437646118700175360 happening it's not a good CI platform anymore.

This PR adds a GitHub Actions config and removes the TravisCI config. You can see this config in action at https://github.com/Turbo87/igc-rs/actions/runs/1238862579.

### CODE REVIEW

**Comment by Joey9801** [comment]
Thanks for the PR! Moving to GH Actions is something I had thought about adding for a while but never really got around to doing.

It looks GH is struggling to deduce that you still want `.travis.yml` deleted after I merged #46 🤷

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - run: cargo check --all-targets --all-features
+        env:
+          RUSTFLAGS: "-D warnings"
+
+  wasm:
+    name: Check (WASM)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: rustup target add wasm32-unknown-unknown
```
Comment (Joey9801) [inline .github/workflows/ci.yml:45]: (Caveat: I haven't written any Rust GH Actions configs myself yet, so don't really know what I'm doing)
It looks like https://github.com/actions-rs/toolchain might be a cleaner way to specify Rust versions / targets.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+
+jobs:
```
Comment (Joey9801) [inline .github/workflows/ci.yml:14]: We should probably explicitly include `stable` and the MSRV in the checks - I'm not sure what version ships in the `ubuntu-latest` image.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Turbo87) [reply .github/workflows/ci.yml:45]: yeah, I had used that action at first on several projects, but I never noticed a significant advantage over just using rustup directly. and given that the last update of that action was 11 months ago I'm also a bit worried about the long-term maintenance of it.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Turbo87) [reply .github/workflows/ci.yml:14]: yep, that makes sense now that #46 is merged 😅

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Joey9801) [reply .github/workflows/ci.yml:45]: That's a good point - I just had a peek at its source code, and all it does is call `rustup` for you 😞

**Comment by Turbo87** [comment]
I had to bump the MSRV once more due to our transitive dependency on https://crates.io/crates/remove_dir_all :-/

**Comment by Turbo87** [comment]
@Joey9801 any thoughts on these changes?

**Comment by Turbo87** [comment]
🤷‍♂️

**Pull request closed without merging**